src/main/java/ui/components/pickers/MilestonePicker.java

```diff
@@ -0,0 +1,44 @@
+package ui.components.pickers;
+
+import backend.resource.TurboIssue;
+import backend.resource.TurboMilestone;
+import javafx.application.Platform;
+import javafx.scene.control.ButtonType;
+import javafx.stage.Stage;
+import javafx.util.Pair;
+import ui.UI;
+import undo.actions.ChangeMilestoneAction;
+import util.events.ShowMilestonePickerEventHandler;
+
+import java.util.List;
+import java.util.Optional;
+
+public class MilestonePicker {
+    private final UI ui;
+    private final Stage stage;
+
+    public MilestonePicker(UI ui, Stage mainStage) {
+        this.ui = ui;
+        this.stage = mainStage;
+        ui.registerEvent((ShowMilestonePickerEventHandler) e -> Platform.runLater(() -> showMilestonePicker(e.issue)));
+    }
+
+    private void showMilestonePicker(TurboIssue issue) {
+        List<TurboMilestone> milestones = ui.logic.getRepo(issue.getRepoId()).getMilestones();
+        MilestonePickerDialog dialog = new MilestonePickerDialog(stage, issue, milestones);
+        Optional<Pair<ButtonType, Optional<Integer>>> milestoneDialogResponse = dialog.showAndWait();
+
+        if (!milestoneDialogResponse.isPresent() ||
+                milestoneDialogResponse.get().getKey().equals(ButtonType.CANCEL)) {
```

Comment:
This method feels like it could be a bit more SLAP. We can use an inner class instead of the tuple, with a method `wasCancelled` maybe?

The line above can be

```java
Optional<MilestonePickerDialog.Response> response =
    new MilestonePickerDialog(stage, issue, milestones).showAndWait();
```
to cut as much noise as possible.

Registering the undo action also seems like it could be pulled out into a method.